doc/main/hubs/essentialhub.rst

```diff
@@ -0,0 +1,147 @@
+.. pybricks-requirements:: primehub
+
+Essential Hub
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. figure:: ../../main/images/primehub.png
+    :height: 15 em
+
+.. autoclass:: pybricks.hubs.EssentialHub
+    :no-members:
+
+    .. rubric:: Using the hub status light
+
+    .. figure:: ../../main/images/primehub_light_label.png
+        :width: 22 em
+
+    .. automethod:: pybricks.hubs::EssentialHub.light.on
+
+    .. automethod:: pybricks.hubs::EssentialHub.light.off
+
+    .. automethod:: pybricks.hubs::EssentialHub.light.blink
+
+    .. automethod:: pybricks.hubs::EssentialHub.light.animate
+
+    .. rubric:: Using the button
+
+    .. figure:: ../../main/images/primehub_buttons_label.png
+        :width: 22 em
+
+    .. automethod:: pybricks.hubs::EssentialHub.buttons.pressed
+
+    .. rubric:: Using the IMU
+
+    .. automethod:: pybricks.hubs::EssentialHub.imu.up
+
+    .. automethod:: pybricks.hubs::EssentialHub.imu.tilt
+
+    .. automethod:: pybricks.hubs::EssentialHub.imu.acceleration
+
+    .. automethod:: pybricks.hubs::EssentialHub.imu.angular_velocity
+
+    .. automethod:: pybricks.hubs::EssentialHub.imu.heading
+
+    .. automethod:: pybricks.hubs::EssentialHub.imu.reset_heading
+
+    .. rubric:: Using the battery
+
+    .. automethod:: pybricks.hubs::EssentialHub.battery.voltage
+
+    .. automethod:: pybricks.hubs::EssentialHub.battery.current
+
+    .. rubric:: Getting the charger status
+
+    .. automethod:: pybricks.hubs::EssentialHub.charger.connected
+
+    .. automethod:: pybricks.hubs::EssentialHub.charger.current
+
+    .. automethod:: pybricks.hubs::EssentialHub.charger.status
+
+    .. rubric:: System control
+
+    .. automethod:: pybricks.hubs::EssentialHub.system.set_stop_button
+
+    .. automethod:: pybricks.hubs::EssentialHub.system.name
+
+    .. automethod:: pybricks.hubs::EssentialHub.system.shutdown
+
+    .. automethod:: pybricks.hubs::EssentialHub.system.reset_reason
+
+Status light examples
+---------------------
+
+Turning the light on and off
+****************************
+
+.. literalinclude::
+    ../../../examples/pup/hub_primehub/light_off.py
```

Comment:
It looks like there is a `hub_shared` folder to help generate these automatically.

Reply:
I see, will have a look. Thanks!

Reply:
We could probably make this more obvious by including a header comment line in the generated code that says `# DO NOT EDIT: this file was generated by hub_shared/make_shared_examples.py`. Then use the `:lines:` option of the `..literalinclude::` directive to skip displaying that line in the docs. (sounds like another PR :smile:).

Reply:
Had a look at the `hub_shared`. As crazy as it seems it is not referred to by anything 😆

Every hub just duplicates the same stuff. For example `hub_movehub/button_single.py` and `hub_technichub/button_single.py`

Reply:
Common hub examples are placed in the shared folder, and the hub specific variants are generated from it by running the script in that folder. 

Some might call it crazy, others might call it simple. Not every simple problem needs a complicated solution.

Adding a special extra line only to add even more  code to remove it later does not seem very helpful. 

I would rather not commit generated files (same for images with labels), but that's just something we agreed to in the past.

But we could change that and just make that part of the build process.

Reply:
Generated images are different because of the inkscape dependency (it can be a pain to install and get running on some OSes and even then, different installs of inkscape give different results of the generated files because of different available fonts, etc.). But Python is already a build dependency, so generating those files during a build is fine (as long as no makefile is involved).

Reply:
Oh, now it makes sense! I called it crazy because I haven't found any occurrence of `hub_shared` using full-text-search.